backend/src/main/kotlin/org/loculus/backend/service/submission/SubmissionDatabaseService.kt

```diff
@@ -1276,7 +1273,15 @@ class SubmissionDatabaseService(
         }
     }
 
-    fun useNewerProcessingPipelineIfPossible(): Map<String, Long?> {
+
+    private var lastPreprocessedDataUpdate: String? = null
+
+    /** Check if there have been newly preprocessed data since the last update check.
+     * Returns empty map if no changes to the prepro table since last check.
+     * If there are changes, runs useNewerProcessingPipelineIfPossible for each organism
+     * and returns a map of organism name to newly set pipeline version (or null if no new version was set).
+     */
+    fun runPipelineUpgradeIfPreproTableChanged(): Map<String, Long?> {
```

Comment:
Thanks for the PR - I agree it's confusing as is and this is an improvement


```suggestion
    fun runPotentialPipelineUpgradeIfPreproTableChanged(): Map<String, Long?> {
```
I think it's kinds confusing as proposed that the top function implies a definite upgrade and the subfunction suggests its possible